pkg/utils/policy/policy_test.go

```diff
@@ -0,0 +1,212 @@
+package policy
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	gwv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+	gwv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+
+	mockclient "github.com/aws/aws-application-networking-k8s/mocks/controller-runtime/client"
+	anv1alpha1 "github.com/aws/aws-application-networking-k8s/pkg/apis/applicationnetworking/v1alpha1"
+	"github.com/aws/aws-application-networking-k8s/pkg/model/core"
+)
+
+func Test_checkGatewayOrRouteTargetRefExistsForPolicy(t *testing.T) {
+	testNamespace := gwv1beta1.Namespace("test-ns")
+
+	notFoundError := fmt.Errorf("k8s resource not found")
+	type args struct {
+		policy core.Policy
+	}
+	tests := []struct {
+		name                         string
+		args                         args
+		expectedK8sClientReturnedObj client.Object
+		want                         bool
+		wantErr                      error
+	}{
+		{
+			name: "AccessLogPolicy TargetRef gateway exists in the k8s, happy path",
```

Comment:
nit: Name should describe expected outcome, rather than just say happy path

Reply:
Same for all other happy path cases

Reply:
Should I do "AccessLogPolicy TargetRef gateway exists in the k8s,  expect return true"? but 
```
want:    true,
wantErr: nil,
```
already indicate the expected outcome?

Reply:
Yes, the name of the test should be indicative of what you want from the test, regardless of params. Reason is that if a test fails, I want to read the name of the test in the output and know why it failed without having to read a bunch of code